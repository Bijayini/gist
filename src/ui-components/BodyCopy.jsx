import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import theme from '../lib/theme';

const BodyCopy = ({
  xs,
  xm,
  xl,
  children,
  variant,
  color,
  className,
  uppercase,
  styles,
}) => {
  const variantFor = () => {
    let variantStyles = `font-family: Lato, Sans-Serif;`;
    switch (variant) {
      case 'extra-bold':
        variantStyles += `font-weight: 900`;
        break;
      case 'regular':
        variantStyles += `font-weight: normal`;
        break;
      case 'light':
        variantStyles += `font-weight: 300`;
        break;
      case 'bold':
        variantStyles += `font-weight: 700`;
        break;
      case 'italic':
        variantStyles += `font-style: italic`;
        break;
      default:
        break;
    }
    return variantStyles;
  };

  function fontSize(level) {
    let fontSizes = '1rem';
    switch (level) {
      case '1':
        fontSizes = '1.125rem';
        break;
      case '2':
        fontSizes = '1rem';
        break;
      case '3':
        fontSizes = '0.875rem';
        break;
      case '4':
        fontSizes = '0.75rem';
        break;
      case '5':
        fontSizes = '0.5rem';
        break;
      case 'large':
        fontSizes = '1.25rem';
        break;
      case 'x-large':
        fontSizes = '1.375rem';
        break;
      default:
        fontSizes = '1rem';
        break;
    }
    return fontSizes;
  }

  return (
    <Fragment>
      <span
        className={`body-copy-${color}-${variant} body-xm body-xs body-xl ${
          className.length ? className : ''
        }`}
      >
        {children}
      </span>
      <style jsx="true">
        {`
          span {
           ${uppercase ? 'text-transform: uppercase;' : ''}
          }
          .body-copy-${color}-${variant} {
            ${variantFor()};
            color: ${theme[color]};
          }
          .body-xs {
            font-size: ${fontSize(xs)}
          }
          @media screen and (${theme.breakpoints.tablet}) {
            .body-xm {
              font-size: ${fontSize(xm)}
            }
          }
          @media screen and (${theme.breakpoints.desktop}) {
            .body-xl {
              font-size: ${fontSize(xl)}
            }
          }
        `}
      </style>
      {styles}
    </Fragment>
  );
};

BodyCopy.defaultProps = {
  xs: '',
  xm: '',
  xl: '',
  className: '',
  uppercase: false,
  styles: false,
};

BodyCopy.propTypes = {
  children: PropTypes.node.isRequired,
  /** Level 1..4 for mobile */
  xs: PropTypes.string,
  /** Level 1..4 for tablet */
  xm: PropTypes.string,
  /** Level 1..4 for desktop */
  xl: PropTypes.string,
  variant: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  className: PropTypes.string,
  uppercase: PropTypes.bool,
  styles: PropTypes.node,
};

export default BodyCopy;
