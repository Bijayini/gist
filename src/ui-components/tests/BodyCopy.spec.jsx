import React from 'react';
import { shallow } from 'enzyme';

import BodyCopy from '../BodyCopy';

describe('<BodyCopy />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      variant: '',
      color: 'white',
      className: 'random',
    };
  });

  it('should match with snapshot with level 3 and italic variant', () => {
    props.xs = '3';
    component = shallow(<BodyCopy {...props}> Hello </BodyCopy>);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should match with snapshot with level 1 and extra-bold variant', () => {
    props.xs = '1';
    props.variant = 'extra-bold';
    component = shallow(<BodyCopy {...props}> Hello </BodyCopy>);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should match with snapshot with level 2 and regular variant', () => {
    props.xs = '2';
    props.variant = 'regular';
    component = shallow(<BodyCopy {...props}> Hello </BodyCopy>);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should match with snapshot with level 4 and light variant', () => {
    props.xs = '4';
    props.variant = 'light';
    component = shallow(<BodyCopy {...props}> Hello </BodyCopy>);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should match with snapshot with level large and bold variant', () => {
    props.xs = 'large';
    props.variant = 'bold';
    component = shallow(<BodyCopy {...props}> Hello </BodyCopy>);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should match with snapshot with level 5 and italic variant', () => {
    props.xs = '5';
    props.variant = 'italic';
    component = shallow(<BodyCopy {...props}> Hello </BodyCopy>);
    expect(component.debug()).toMatchSnapshot();
  });
});
