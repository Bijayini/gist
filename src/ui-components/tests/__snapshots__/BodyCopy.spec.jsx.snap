// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyCopy /> should match with snapshot with level 1 and extra-bold variant 1`] = `
"<Fragment>
  <span className=\\"jsx-2222279394 body-copy-white-extra-bold body-xm body-xs body-xl random\\">
     Hello 
  </span>
  <JSXStyle id=\\"742731879\\" dynamic={{...}}>
    span.__jsx-style-dynamic-selector{;}.body-copy-white-extra-bold.__jsx-style-dynamic-selector{font-family: Lato, Sans-Serif;font-weight: 900;color:#ffffff;}.body-xs.__jsx-style-dynamic-selector{font-size:1.125rem;}@media screen and (min-width: 768px){.body-xm.__jsx-style-dynamic-selector{font-size:1rem;}}@media screen and (min-width: 1248px){.body-xl.__jsx-style-dynamic-selector{font-size:1rem;}}
  </JSXStyle>
</Fragment>"
`;

exports[`<BodyCopy /> should match with snapshot with level 2 and regular variant 1`] = `
"<Fragment>
  <span className=\\"jsx-462673144 body-copy-white-regular body-xm body-xs body-xl random\\">
     Hello 
  </span>
  <JSXStyle id=\\"742731879\\" dynamic={{...}}>
    span.__jsx-style-dynamic-selector{;}.body-copy-white-regular.__jsx-style-dynamic-selector{font-family: Lato, Sans-Serif;font-weight: normal;color:#ffffff;}.body-xs.__jsx-style-dynamic-selector{font-size:1rem;}@media screen and (min-width: 768px){.body-xm.__jsx-style-dynamic-selector{font-size:1rem;}}@media screen and (min-width: 1248px){.body-xl.__jsx-style-dynamic-selector{font-size:1rem;}}
  </JSXStyle>
</Fragment>"
`;

exports[`<BodyCopy /> should match with snapshot with level 3 and italic variant 1`] = `
"<Fragment>
  <span className=\\"jsx-349263200 body-copy-white- body-xm body-xs body-xl random\\">
     Hello 
  </span>
  <JSXStyle id=\\"742731879\\" dynamic={{...}}>
    span.__jsx-style-dynamic-selector{;}.body-copy-white-.__jsx-style-dynamic-selector{font-family: Lato, Sans-Serif;;color:#ffffff;}.body-xs.__jsx-style-dynamic-selector{font-size:0.875rem;}@media screen and (min-width: 768px){.body-xm.__jsx-style-dynamic-selector{font-size:1rem;}}@media screen and (min-width: 1248px){.body-xl.__jsx-style-dynamic-selector{font-size:1rem;}}
  </JSXStyle>
</Fragment>"
`;

exports[`<BodyCopy /> should match with snapshot with level 4 and light variant 1`] = `
"<Fragment>
  <span className=\\"jsx-3020890833 body-copy-white-light body-xm body-xs body-xl random\\">
     Hello 
  </span>
  <JSXStyle id=\\"742731879\\" dynamic={{...}}>
    span.__jsx-style-dynamic-selector{;}.body-copy-white-light.__jsx-style-dynamic-selector{font-family: Lato, Sans-Serif;font-weight: 300;color:#ffffff;}.body-xs.__jsx-style-dynamic-selector{font-size:0.75rem;}@media screen and (min-width: 768px){.body-xm.__jsx-style-dynamic-selector{font-size:1rem;}}@media screen and (min-width: 1248px){.body-xl.__jsx-style-dynamic-selector{font-size:1rem;}}
  </JSXStyle>
</Fragment>"
`;

exports[`<BodyCopy /> should match with snapshot with level 5 and italic variant 1`] = `
"<Fragment>
  <span className=\\"jsx-3378258428 body-copy-white-italic body-xm body-xs body-xl random\\">
     Hello 
  </span>
  <JSXStyle id=\\"742731879\\" dynamic={{...}}>
    span.__jsx-style-dynamic-selector{;}.body-copy-white-italic.__jsx-style-dynamic-selector{font-family: Lato, Sans-Serif;font-style: italic;color:#ffffff;}.body-xs.__jsx-style-dynamic-selector{font-size:0.5rem;}@media screen and (min-width: 768px){.body-xm.__jsx-style-dynamic-selector{font-size:1rem;}}@media screen and (min-width: 1248px){.body-xl.__jsx-style-dynamic-selector{font-size:1rem;}}
  </JSXStyle>
</Fragment>"
`;

exports[`<BodyCopy /> should match with snapshot with level large and bold variant 1`] = `
"<Fragment>
  <span className=\\"jsx-347687786 body-copy-white-bold body-xm body-xs body-xl random\\">
     Hello 
  </span>
  <JSXStyle id=\\"742731879\\" dynamic={{...}}>
    span.__jsx-style-dynamic-selector{;}.body-copy-white-bold.__jsx-style-dynamic-selector{font-family: Lato, Sans-Serif;font-weight: 700;color:#ffffff;}.body-xs.__jsx-style-dynamic-selector{font-size:1.25rem;}@media screen and (min-width: 768px){.body-xm.__jsx-style-dynamic-selector{font-size:1rem;}}@media screen and (min-width: 1248px){.body-xl.__jsx-style-dynamic-selector{font-size:1rem;}}
  </JSXStyle>
</Fragment>"
`;
